app/controllers/api/events_controller.rb

```diff
@@ -9,13 +9,24 @@ def mark_as_read
   private
 
   def accessible_records
-    load_and_authorize(:discussion).items.sequenced
+    if params[:event_id_gt]
+      Event.joins(:events_groups).
+        where("events_groups.group_id in (?)", current_user.group_ids).
+        where("events.id > ?", params[:event_id_gt]).order(:id)
+    else
+      load_and_authorize(:discussion).items.sequenced
+    end
   end
 
   def page_collection(collection)
```

Comment:
sorry @gdpelican. please show me how to this better.

Reply:
This is pointing towards a separate endpoint for me. This new endpoint is more like `events#index`, whereas the existing discussion endpoint is more like `events#for_discussion`

```
def index
  # I'd strongly consider a timestamp here; this feels more like "Give me all the events which happened since I disconnected at <time>"
  instantiate_collection { |collection| collection.where("events.id > ?", params[:event_id]) }
  respond_with_collection
end

def for_discussion
  instantiate_collection do |collection|
    collection.where(discussion: load_and_authorize(:discussion)).sequenced
  end
  respond_with_collection
end

private

def accessible_records
  resource_class.joins(:events_groups)
                .where("events_groups.group_id": current_user.group_ids)
end
```